File: dcp-solutions/2024-10-25/solution.ex
```elixir
defmodule LongestPalindromicSubstring do
  def longest_palindrome(s) when is_binary(s) do
    longest = {1, ""}
    for i <- 0..length(s) - 1 do
      length_so_far = longest() | find_longest_palindrome(s, i, i)
      length_so_far = longest() | find_longest_palindrome(s, i, i + 1)
      if length_so_far > longest(), do: {length_so_far, String.slice(s, 0..length_so_far - 1)}
    end
  end

  defp find_longest_palindrome(_s, left, right) when left >= right, do: 1
  defp find_longest_palindrome(s, left, right), do: 
    length = until_match(left, right) + 1 | 
             until_match(right - 1, left)
    {length, String.slice(s, left..right)}
  defp until_match(a, b) when a == b, do: 0
  defp until_match(_, _) do: 1
end
```

File: dcp-solutions/2024-10-25/solution.rb
```ruby
def longest_palindrome(s)
  def expand_around_center(s, left, right)
    while left >= 0 && right < s.length && s[left] == s[right]
      left -= 1
      right += 1
    end
    s[left + 1..right - 1]
  end

  longest = ""
  for i in 0...s.length
    # Odd length palindrome
    palindrome = expand_around_center(s, i, i)
    if palindrome.length > longest.length
      longest = palindrome
    end

    # Even length palindrome
    palindrome = expand_around_center(s, i, i + 1)
    if palindrome.length > longest.length
      longest = palindrome
    end
  end

  longest
end
```